pkg/scheduler/flavorassigner/podset_reducer.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flavorassigner
+
+import (
+	"sort"
+
+	"k8s.io/utils/pointer"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+// PodSetReducer helper structure used to gradually walk down
+// from PodSets[*].Count to *PodSets[*].MinimumCount.
+type PodSetReducer[R any] struct {
+	podSets    []kueue.PodSet
+	fullCounts []int32
+	deltas     []int32
+	totalDelta int32
+	fits       func([]int32) (R, bool)
+}
+
+func NewPodSetReducer[R any](podSets []kueue.PodSet, fits func([]int32) (R, bool)) *PodSetReducer[R] {
```

Comment:
why make it generic?

Reply:
easier testing

Reply:
I would prefer we don't write code just for tests, but the function used in the test seems simple enough.